app/src/lib/stores/app-store.ts

```diff
@@ -4129,12 +4129,33 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
   public async _undoCommit(
     repository: Repository,
-    commit: Commit
+    commit: Commit,
+    showConfirmationDialog: boolean
   ): Promise<void> {
     const gitStore = this.gitStoreCache.get(repository)
+    const repositoryState = this.repositoryStateCache.get(repository)
+    const { changesState } = repositoryState
+
+    if (showConfirmationDialog) {
+      // Compare file paths in working directory and in the commit that will be undone
+      const workingDirectoryFiles = changesState.workingDirectory.files.map(
+        f => f.path
+      )
+      const commitFiles = (await getChangedFiles(repository, commit.sha)).map(
+        f => f.path
+      )
+
+      // Warn the user if there is any overlap between commit files and working
+      // directory files
```

Comment:
@billygriffin I kept this behavior that we discussed when the idea was to suggest stashing the changes. Does it still make sense to you? Or should we warn the user even when the current changes don't overlap the commit changes?